backends/edk2-compat/edk2-svc-validate.c

```diff
@@ -274,51 +273,55 @@ int validateAuth(const unsigned char *authBuf, size_t buflen, const char *key)
  */
 int validatePKCS7(const unsigned char *cert_data, size_t len) 
 {
-	mbedtls_x509_crt *pkcs7cert = NULL;
-	mbedtls_pkcs7 *pkcs7 = NULL;
-	int rc;
+	void *pkcs7 = NULL, *pkcs7_cert = NULL;
+	int rc = SUCCESS, cert_num = 0;
 
 	prlog(PR_INFO, "VALIDATING PKCS7:\n");
-	pkcs7 = malloc(sizeof(struct mbedtls_pkcs7));
-	if (!pkcs7){
-		prlog(PR_ERR, "ERROR: failed to allocate memory\n");
-		return ALLOC_FAIL;
-	}
-	mbedtls_pkcs7_init(pkcs7);
-	rc = mbedtls_pkcs7_parse_der(cert_data, len, pkcs7);
-	if (rc != MBEDTLS_PKCS7_SIGNED_DATA) {	// if pkcs7 parsing fails, then try new signed data format 
-			prlog(PR_ERR, "ERROR: parsing pkcs7 failed mbedtls error #%04x\n", rc);
-			goto out;	
+	// pkcs7 = malloc(crypto_get_pkcs7_struct_size() );
+	// if (!pkcs7){
+	// 	prlog(PR_ERR, "ERROR: failed to allocate memory\n");
+	// 	return ALLOC_FAIL;
+	// }
```

Comment:
Delete commented out large segments unless they are absolutely still needed for reference. You can always look back at the commit history, or check out an earlier commit/branch if you need to look at it again.